sim/pkg/model/autoscaler_ticktock.go

```diff
@@ -98,13 +109,55 @@ func (asts *autoscalerTicktockStock) Add(entity simulator.Entity) error {
 	} else {
 		// do nothing
 	}
+}
 
-	//calculate CPU utilization
-	asts.calculateCPUUtilization()
+func (asts *autoscalerTicktockStock) adjustVertically(currentTime *time.Time) {
+	recommendedPodResources, err := asts.env.Plugin().VerticalRecommendation(currentTime.UnixNano())
+	if err != nil {
+		panic(err)
+	}
+	//create a map podName -> []recommendedPodResources
+	podToRecommendations := make(map[string][]*proto.RecommendedPodResources, 0)
+	for _, recommendation := range recommendedPodResources {
+		podToRecommendations[recommendation.PodName] = append(podToRecommendations[recommendation.PodName], recommendation)
+	}
```

Comment:
Does VPA provide recommendations for each pod?  I thought it was per container.  So if a Deployment template has pod with 2 containers, VPA would provide 2 recommendations.  No matter how many pods there are in the deployment.

@bskiba to confirm.

I see this code is consistent with the API: https://github.com/pivotal/skenario/blob/master/plugin/pkg/skplug/proto/skplug.proto#L61-L67 where I would expect `pod` to be `container`.  Sorry I didn't catch this earlier.

Reply:
Yes, VPA recommends per container. However, around the design phase there was an assumption that we simplify and assume always 1 container per pod (https://docs.google.com/document/d/1bD6q32JrqPetZniOe-VMYkm29Hh8-WhNEyCIOipD3TY/edit?disco=AAAAJ-e3U_I)

Reply:
To be clear, if a Deployment has 100 pods, then VPA will produce 100 recommendations?

Reply:
No, just one per deployment.